### DISCUSSION TITLE
Goto pytest fixtures definitions

### DISCUSSION THREAD

**Discussion by kosciej16** [microsoft/pyright#2117] [Q&A]
Hi, I use Coc-pyright to integrate pyright into vim and it works great, but I miss one functionality a lot - **I cannot goto definition of pytest fixture** (located in some conftest.py).
With jedi it works, do I do something wrong or such goto is not supported by pyright? If not, do you have any recomendation?

Thanks for help!

**Comment by erictraut** [comment]
If the pytest fixture is not explicitly imported but rather implicitly referenced by the conftest.py, then pyright has no knowledge of it. Perhaps Jedi has specific knowledge of pytest and its configuration files. Pyright doesn't have knowledge of any specific third-party libraries, and we have no plans to add such knowledge. The solution in this case is to explicitly import fixtures in your test files rather than relying on pytest's dynamic behaviors.

**Reply by kosciej16** [reply]
Thank you @erictraut for an answer! As importing fixture has little sense (it should not be fixture then!) I will think about use both and combining pros of them.

**Reply by r3m0t** [reply]
Could you annotate type of fixture?
``def test(myfixture: MyFixtureType): ...``

Then use gotodefinition on MyFixtureType, which is defined in conftest.py using https://docs.python.org/3/library/typing.html#typing.NewType

**Reply by kosciej16** [reply]
Sorry for late answer - yes, I can annotate that but we rarely write code from scratch. Usually there are hundreds of tests already written and they have no such annotations so it's not a solution.
